### PR TITLE
Project To-do List - Ellen

### PR DESCRIPTION
**Pull request by ellenbjo**
Link to my deployed project:
https://projecttodos.netlify.app/
My reflections on how this week's project turned out:
This has been a really messy week and I've been struggling with different bugs and things not working as I want them to. The functionality is there, but the styling is not done yet. Will come back to the styling part later!
However I got a much bette understanding of redux and reducers :)

Things I'd like to have clarified or explained in more detail:
local storage. I'm using it, but not 100% sure on how it works.

### CODE REVIEW

**Review by lilihorvath — COMMENTED** [review]
Dear Ellen! First of all, sorry for the late review! You made an incredibly good job on this project even that your week was messy. I am still struggling with mine.I am still learning how to use correctly the local storage in this project, but as I saw you managed it really well! You code is simple, easy to read, easily can understand the components and the names of them. I like how you did the styled-component part separate files, easy to find too! Please forgive me for this short code review: as I mentioned I am still on my way to understand clearly the issues of this week's challenges. You did a very good job!

File: code/src/components/DeleteAll.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import { DeleteAllButton } from '../styledComponents/buttons'
+
+export const DeleteAll = () => {
+  const dispatch = useDispatch()
+
+  const onRemoveClick = () => {
+    dispatch(todos.actions.removeAll())
+    dispatch(
+      todos.actions.showList()
+    )
```
Comment (HIPPIEKICK) [inline code/src/components/DeleteAll.js]: Consistency:
```
dispatch(todos.actions.removeAll())
dispatch(todos.actions.showList())
```

File: code/src/components/EmptyList.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import React from 'react'
+
+import { Text } from '../styledComponents/texts'
+import { EmptyListContainer } from '../styledComponents/containers'
+
+export const EmptyList = () => {
+  return (
+    <EmptyListContainer>
+      <Text>No to-dos yet!</Text>
```
Comment (HIPPIEKICK) [inline code/src/components/EmptyList.js:9]: Nice with an empty-state!

File: code/src/components/Header.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+import { Title } from '../styledComponents/texts'
+import { HeaderSection } from '../styledComponents/containers'
+
+export const Header = () => {
+  return (
+    <HeaderSection>
+      <Title>TO-DO LIST</Title>
+    </HeaderSection>
+  )
+}
```
Comment (HIPPIEKICK) [inline code/src/components/Header.js]: Convention: one empty line at the end of each file

File: code/src/components/Input.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+            <InputField
+            type='text'
+            onChange={event => setInputValue(event.target.value)}
+            value={inputValue}
+            placeholder="Walk the dog"
+            />
+            <AddButton
+              type='submit'
+              value='add todo'
+              disabled={inputValue.length < 1}
```
Comment (HIPPIEKICK) [inline code/src/components/Input.js:46]: Visualize this to the user so that they know they can't submit

File: code/src/components/Summary.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import React from 'react'
+import { useSelector } from 'react-redux'
+
+import { Text } from '../styledComponents/texts'
+
+export const Summary = () => {
+  const list = useSelector((store) => store.todos.list)
```
Comment (HIPPIEKICK) [inline code/src/components/Summary.js]: If you only ned the todoItems array you can get only that

File: code/src/components/List.js [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    )
+  } else {
+    return (
+      <ListContainer>
+        {todoItems.map((item) => (
+          <Item key={item.id} item={item}/>
+        ))}
+      </ListContainer>
+    )
+  }
```
Comment (HIPPIEKICK) [inline code/src/components/List.js]: Do you need both this...

File: code/src/pages/Home.js [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+
+  return(
+    <MainContainer>
+      <Header />
+      <TopContainer>
+        <DateInfo />
+        <Summary />
+      </TopContainer>
+      {!showList && <EmptyList />}
+      {showList && <List />}
```
Comment (HIPPIEKICK) [inline code/src/pages/Home.js:23]: And this?

File: code/src/components/DeleteAll.js [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+  return (
+    <Wrapper>
+      <Text>
+        Delete All
+      </Text>
+      <DeleteAllButton type='button' onClick={onRemoveClick}>
+        X
+      </DeleteAllButton>
+    </Wrapper>
+  )
```
Comment (HIPPIEKICK) [inline code/src/components/DeleteAll.js:37]: Would be nice if this component were only visible if there are todos

**Pull request closed without merging**